src/Slices/ServiceDetails/UI/Tabs/Callbacks/CreateCallbackForm.tsx

```diff
@@ -8,49 +8,71 @@ import {
+import { LogLevelsList, EventTypesList, LogLevelString, EventType } from "@/Core";
+import { useCreateCallback } from "@/Data/Managers/V2/Callback";
 import { MultiTextSelect, SingleTextSelect } from "@/UI/Components";
-import { DependencyContext } from "@/UI/Dependency";
 import { words } from "@/UI/words";
 
 interface Props {
   service_entity: string;
   numberOfColumns: number;
 }
 
+/**
+ * CreateCallbackForm component
+ * Provides a form for creating new callbacks.
+ * It allows users to input callback URL, ID, log level, and event types, and handles the creation
+ * of callbacks through the API.
+ *
+ * @props {Props} props - The props of the component.
+ * @prop {string} service_entity  - The entity associated with the callback
+ * @prop {number} numberOfColumns - The number of columns in the parent table for proper layout
+ *
+ * @returns {React.FC<Props>} The rendered Component to create a callback.
+ */
 export const CreateCallbackForm: React.FC<Props> = ({ service_entity, numberOfColumns }) => {
+  const client = useQueryClient();
   const [url, setUrl] = useState<string | null>(null);
   const [id, setId] = useState<string | null>(null);
   const [logLevel, setLogLevel] = useState<string | null>(null);
   const [eventTypes, setEventTypes] = useState<string[]>([]);
   const [error, setError] = useState<string | null>(null);
-  const { commandResolver } = useContext(DependencyContext);
   const logOptions: SelectOptionProps[] = LogLevelsList.map((option) => {
     return { value: option, children: option };
   });
 
-  const create = commandResolver.useGetTrigger<"CreateCallback">({
-    kind: "CreateCallback",
-    callback_url: url || "",
-    callback_id: id || undefined,
-    service_entity,
-    minimal_log_level: logLevel === null ? undefined : (logLevel as LogLevelString),
-    event_types: eventTypes.length <= 0 ? undefined : (eventTypes as EventType[]),
-  });
-
-  const onCreate = async () => {
-    setError(null);
-    const error = await create();
-
-    if (Maybe.isSome(error)) {
-      setError(error.value);
-    } else {
+  const { mutate } = useCreateCallback({
+    onError: (error) => {
+      setError(error.message);
+    },
+    onSuccess: () => {
+      //invalidate the get_callbacks query to update the list
+      client.invalidateQueries({
+        queryKey: ["get_callbacks-one_time"],
+      });
       setUrl(null);
       setId(null);
       setLogLevel(null);
       setEventTypes([]);
-    }
+    },
+  });
+
+  /**
+   * Handles the creation of a new callback.
+   * This function gather the input data and calls the mutation function to create a callback.
+   * It resets the error state before attempting to create the callback.
+   */
+  const onCreate = () => {
+    setError(null);
+    mutate({
+      callback_url: url || "",
+      callback_id: id || undefined,
+      service_entity,
+      minimal_log_level: logLevel === null ? undefined : (logLevel as LogLevelString),
```

Comment:
Why do we need the casting here?

Reply:
because select box convert logLevel from `LogLevelString` to `string` and to keep typings align I opted for what was already there

Reply:
kay